Word replacements and translation replacements never apply on Standalone.

SettingsManager::bootSettings() loads the settings metadata, whose titles go
through ts(), before CIVICRM_DSN is defined. CRM_Core_I18n caches the empty
replacement list it returns at that point for the rest of the request.

Upstream: https://lab.civicrm.org/dev/core/-/work_items/5862
Fix: https://github.com/civicrm/civicrm-core/pull/37148

diff --git a/CRM/Core/I18n.php b/CRM/Core/I18n.php
index d7c5c7ee48..bbe28b3e93 100644
--- a/CRM/Core/I18n.php
+++ b/CRM/Core/I18n.php
@@ -788,11 +788,15 @@ class CRM_Core_I18n {
     if (defined('CIVI_SETUP') || isset(Civi\Test::$statics['testPreInstall'])) {
       return [];
     }
+    // Standalone translates settings metadata during boot, before the DSN exists: don't cache that.
+    if (!defined('CIVICRM_DSN')) {
+      return [];
+    }
 
     // FIXME: Is there a constant we can reference instead of hardcoding en_US?
     $replacementsLocale = $this->locale ?: 'en_US';
     if ((!isset(Civi::$statics[__CLASS__]) || !array_key_exists($replacementsLocale, Civi::$statics[__CLASS__]))) {
-      if (defined('CIVICRM_DSN') && !CRM_Core_Config::isUpgradeMode()) {
+      if (!CRM_Core_Config::isUpgradeMode()) {
         Civi::$statics[__CLASS__][$replacementsLocale] = CRM_Core_BAO_WordReplacement::getLocaleCustomStrings($replacementsLocale);
       }
       else {
@@ -806,13 +810,17 @@ class CRM_Core_I18n {
     if (defined('CIVI_SETUP') || isset(Civi\Test::$statics['testPreInstall'])) {
       return [];
     }
+    // Standalone translates settings metadata during boot, before the DSN exists: don't cache that.
+    if (!defined('CIVICRM_DSN')) {
+      return [];
+    }
 
     // FIXME: Is there a constant we can reference instead of hardcoding en_US?
     $replacementsLocale = $this->locale ?: 'en_US';
     // temporary to avoid collision with word replacements
     $translationReplacement = 'tr-' . $replacementsLocale;
     if ((!isset(Civi::$statics[__CLASS__]) || !array_key_exists($translationReplacement, Civi::$statics[__CLASS__]))) {
-      if (defined('CIVICRM_DSN') && !CRM_Core_Config::isUpgradeMode() && CRM_Core_BAO_Domain::isDBVersionAtLeast('6.7.beta')) {
+      if (!CRM_Core_Config::isUpgradeMode() && CRM_Core_BAO_Domain::isDBVersionAtLeast('6.7.beta')) {
         Civi::$statics[__CLASS__][$translationReplacement] = CRM_Core_BAO_TranslationSource::getTranslationSources($replacementsLocale);
       }
       else {
